dagql/idtui/frontend_pretty.go

```diff
@@ -742,7 +742,24 @@ func (fe *frontendPretty) update(msg tea.Msg) (*frontendPretty, tea.Cmd) { //nol
 			cmd = &wrapCommand{
 				ExecCommand: cmd,
 				before: func() error {
-					ttyFd := int(os.Stdout.Fd())
+					var in *os.File
+					inr, _ := findTTYs()
+					if inr != nil {
+						in = inr.(*os.File)
+					}
+
+					if in == nil {
+						tty, err := openInputTTY()
```

Comment:
We should avoid opening this again - it should already be open right? 

> Perhaps we can save the result that we pass to `tea.WithInput` onto `frontendPretty` somewhere, and then use that instead?

I think we can save it from the `runWithTUI` function - or is there some reason that we need to find all the TTYs again?

Reply:
no, this is most likely my bad. I'll take a look again at this.

Reply:
I made changes to store the already opened handle in `frontendPretty` struct. I tested with Mac client, it works fine. i will test it with Windows client once as well and report back here.

I also want to look at it with fresh eyes to ensure there is no other optimization we could do for these changes.